impl/maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -1240,7 +1240,9 @@ public String getDefaultGoal() {
     }
 
     public Plugin getPlugin(String pluginKey) {
-        return getBuild().getPluginsAsMap().get(pluginKey);
+        org.apache.maven.api.model.Plugin plugin =
```

Comment:
why do we have both  org.apache.maven.model.Plugin and  org.apache.maven.api.model.Plugin? I'm not sure, but this might be a symptom of too many packages, over-abstraction, or both. Do we need multiple plugin types?

Reply:
We're translating from the Maven 3 api to the Maven 4 api here.  `org.apache.maven.model.Plugin` is the Maven 3 api, with mutable objects.  `org.apache.maven.api.model.Plugin` is the (now underlying) Maven 4 api with immutable objects.
This is somewhat temporary, the plan is to get away with the Maven 3 api in Maven 5.